packages/core/parcel-bundler/src/cli.js

```diff
@@ -214,6 +214,7 @@ async function bundle(main, command) {
 
   command.throwErrors = false;
   command.scopeHoist = command.experimentalScopeHoisting || false;
+  command.hmrHostname = command.hmrHostname || 'localhost';
```

Comment:
This is a breaking change, so not sure if this should have a default

Reply:
That would still be a breaking change, but yes. If you really wanna do it than that would probably be a cleaner solution

Reply:
The problem is: 

https://github.com/parcel-bundler/parcel/blob/c3d99dee452640cd0ffb0e17682e482e8c3abd85/packages/core/parcel-bundler/src/cli.js#L25-L28

It says it defaults to window.location.hostname but this is not true. It is only true, for some `var hostname` later in the code, that is used in the client side code for connecting to the websocket server:

https://github.com/parcel-bundler/parcel/blob/c3d99dee452640cd0ffb0e17682e482e8c3abd85/packages/core/parcel-bundler/src/builtins/hmr-runtime.js#L26

This default does not reach HMRServer.js (obviously). When the time to check the origin comes, hmrHostname is empty, and the check never takes place. And the check, even if it happened, does nothing, as explained before.

hmrHostname needs to have a default somewhere and `normalizeOptions` is the best place. I don't see why it's a breaking change: people using a specified hmrHostname (because they are using some complex setup) will still keep that.

Another option: deprecate hmrHostname like you mentioned in https://github.com/parcel-bundler/parcel/pull/1794#issue-203903881 and https://github.com/parcel-bundler/parcel/pull/1482#issuecomment-395091843. The flags `host` or `hostname` can be used to make the origin check, and as fallback to window.location.hostname in the client.

What do you think ?